### ISSUE TITLE
Can't configure bitocoind to only send tx via Tor but receive clearnet transactions

### ISSUE THREAD

**Issue by ultimatevert** [bitcoin/bitcoin#14692]
There does not seen to be a way to enable onlynet=onion for sending only and regular clearnet for receiving only.

I expected that onlynet=onion would mean I could only accept inbound connections from only onion nodes. This sort of works but I need to manually add nodes and then if a node is not open also to the outside I am in an isolate Bitcoin network.

0.17

Ubuntu 18.04

**Comment by laanwj** [comment]
This is indeed currently not possible.

Mind that transactions are broadcasted over both incoming and outgoing connections.

**Comment by kristapsk** [comment]
This seems like a good idea to me. New configuration option would be needed, for example, "onlybroadcastnet=onion" or something like that.

**Comment by kristapsk** [comment]
Looked at the code, at the first sight this does not seem too hard to implement. So, if there aren't any strong NACK's against the concept, I could work on this.

It seems to me such configuration, where node receives transactions and blocks from both clearnet and Tor, but only broadcasts transtactions via Tor, helps against some of the issues mentioned in [this paper](https://ieeexplore.ieee.org/document/7163022).

**Comment by ultimatevert** [comment]
I agree with what you say and think it would be cool to have this as an extra option. 

I am also think having the option onlybroadcastnet=onion AND onlybroadcastnet=tor would be useful. The tor option would use tor to connect to normal nodes through tor, that won't solve the issue the paper presents but it could help with anonymity for people that use their full node as a wallet.

**Comment by rytido** [comment]
I believe this functionality is possible currently if you just keep your port closed, right? I have the standard 8 outbound connections, but also allow inbound via Tor this way.

**Comment by ultimatevert** [comment]
Although you have  8 standard connections (outgoing) and tor connections (inbound), when your node broadcasts a transaction you will broadcast to all outgoing and inbound (hence broadcast). I desire a set up where I can configure if my node sends its transaction to outgoing connections or inbound connections. I don't think the software can make that distinction though so it is probably easier to have an option that only broadcasts via onion, as someone else mentioned.

One setup is to configure Tor support to the node, the public key will get created, bhasdhsa.onion, as an example and I can give that to other people and I will have inbount connections via that onion service. The problem still exists where I will still broadcast to both onion and non onion.

Another setup is to have onlynet=onion so that my outgoing connections are only to onion services, previous nodes that added me would still be able to connect as my clearnet port is still open/working. Now the final kicker would be a setting such as 'onlybroadcast=onion' which will make my node only broadcast via .onion service outgoing connections but still receive transactions from clearnet via inbound connections. This should also help solve the "Why bitcoin over Tor isn't a good idea" scenario, because your node is still opened to the clearnet.

**Comment by vasild** [comment]
Implemented in https://github.com/bitcoin/bitcoin/pull/29415 for the `sendrawtransaction` RPC. Not for the wallet yet. For the wallet, one could create unsigned PSBT from the GUI and then use the `walletprocesspsbt` and `sendrawtransaction` RPCs.
